### ISSUE TITLE
Gradient-free methods

### ISSUE THREAD

**Issue by gerlero** [patrick-kidger/optimistix#166]
Hi! Would you recommend this package as a base for adding gradient-free (i.e., global, constrained, evolutionary) optimization algorithms? Or do you feel those would be too difficult to implement here and/or are out of scope for `optimistix`?

**Comment by johannahaffner** [comment]
We've been working on adding support for constrained optimisation for a while now, albeit with a focus on local methods that so leverage derivative information.

A gradient-free solver we do have implemented is Nelder-Mead, and we have had other gradient free solvers on our (wish-) list since https://github.com/patrick-kidger/optimistix/issues/3.

What solvers were you thinking of?

**Comment by gerlero** [comment]
@johannahaffner Thanks! I have a basic JAX-compatible implementation of Differential Evolution (currently with a [SciPy-like API](https://docs.scipy.org/doc/scipy-1.16.2/reference/generated/scipy.optimize.differential_evolution.html)). Since `jax.scipy.optimize` isn’t taking new contributions, I was wondering whether `optimistix` (which I'm already a happy user of) would be an appropriate home for this method and if you’d consider it in scope (and whether you'd think it'd be worth the effort).

**Comment by gerlero** [comment]
@johannahaffner This totally answers my question. Thanks again!

**Comment by patrick-kidger** [comment]
I would add, however, that if you have an externally-maintained 'optimistix-evolution' (or whatever name) package, then we'd be very happy to advertise it from our documentation. (C.f. the equinox ecosystem list: https://docs.kidger.site/equinox/awesome-list/)

**Comment by gerlero** [comment]
@patrick-kidger Thanks! I've ended up publishing my implementation as its own package at https://github.com/gerlero/mutax. You're free to add a link to it (let me know if you want me to open a PR; I didn't do it because I'm not sure where on the list to put it).

**Comment by patrick-kidger** [comment]
Awesome! A PR adding it to https://github.com/patrick-kidger/equinox/blob/main/docs/awesome-list.md would be the best place I think. :)
